cpp/src/arrow/type_traits.cc

```diff
@@ -67,21 +67,23 @@ int RequiredValueAlignmentForBuffer(Type::type type_id, int buffer_index) {
     case Type::BINARY:  // Offsets may be cast to int32_t*
     case Type::DATE32:
     case Type::TIME32:
-    case Type::LIST:  // Offsets may be cast to int32_t*, data is in child array
-    case Type::MAP:   // This is a list array
+    case Type::LIST:       // Offsets may be cast to int32_t*, data is in child array
+    case Type::LIST_VIEW:  // Same as LIST
```

Comment:
Offsets and sizes?

Reply:
No need to go into so much detail. Take this to mean that "aligment requirements of numeric buffers are the same between list and list-views".

Reply:
(I've checkout the largelist, and it has, so maybe note here is proper?)

Reply:
Fair enough. Made them all consistent, but dropped the comment about child values from the `LIST` case.